examples/client/Locomotion/src/context/newRideContext/index.js

```diff
@@ -62,7 +64,33 @@ const RidePageContextProvider = ({ navigation, children }) => {
   const [searchResults, setSearchResults] = useState(null);
   const [isReadyForSubmit, setIsReadyForSubmit] = useState(false);
   const [historyResults, setHistoryResults] = useState([]);
-
+  const [serviceEstimations, setServiceEstimations] = useState(null);
+
+  const formatEstimations = (services, estimations) => services.map((service) => {
+    const estimationForService = estimations.find(estimation => estimation.serviceId === service.id);
```

Comment:
Please create a map of {
[estimation.serviceId]: esstimation
} before the services.map And then access it via the map.
better runtime